File: include/PolyWorksSDK/samples/plugins/common/VisualCpp/VCppLineScanFramework/CScanningProfilesUser.cpp
// =================================================================================================
//   CScanningProfilesUser class implementation.
//
// Copyright © InnovMetric Software Inc. 2014 All Rights Reserved
// -------------------------------------------------------------------------------------------------

// =================================================================================================
// ======================================== INCLUDED FILES =========================================

#include "stdafx.h"

#include "CScanningProfilesUser.h"

// =================================================================================================
// ========================================= NAMESPACES ============================================

// =================================================================================================
// =========================================== MACROS ==============================================

// =================================================================================================
// ====================================== CONSTANTS, ENUMS =========================================

// =================================================================================================
// ================================ TYPES, CLASSES, STRUCTURES =====================================

// =================================================================================================
// ================================== STATIC MEMBER VARIABLES ======================================

// =================================================================================================
// ===================================== GLOBAL VARIABLES ==========================================

// =================================================================================================
// ===================================== FUNCTION PROTOTYPES =======================================

// =================================================================================================
// ===================================== FUNCTION DEFINITIONS ======================================

// =================================================================================================
//   Ctor
// -------------------------------------------------------------------------------------------------
CScanningProfilesUser::CScanningProfilesUser()
{
}

// =================================================================================================
//   Dtor
// -------------------------------------------------------------------------------------------------
CScanningProfilesUser::~CScanningProfilesUser()
{
}

// =================================================================================================
//   Gets available scanning profiles.
//
// Parameter : profileCmdArgNames_    : Profile names used for command arguments
// Parameter : profileDisplayNames_   : Profile display names
// Parameter : profileAddtionalInfos_ : Profile additional information
// Parameter : currentProfile_        : Current profile
// -------------------------------------------------------------------------------------------------
void CScanningProfilesUser::ScanningProfilesGet( TWStrings&    profileCmdArgNames_,
                                                 TWStrings&    profileDisplayNames_,
                                                 TWStrings&    profileAddtionalInfos_,
                                                 std::wstring& currentProfile_ ) const
{
    profileCmdArgNames_    = m_profileCmdArgNames;
    profileDisplayNames_   = m_profileDisplayNames;
    profileAddtionalInfos_ = m_profileAddtionalInfos;
    currentProfile_        = m_currentProfile;
}

// =================================================================================================
//   Sets available scanning profiles.
//
// Parameter : profileCmdArgNames_    : Profile names used for command arguments
// Parameter : profileDisplayNames_   : Profile display names
// Parameter : profileAddtionalInfos_ : Profile additional information
// Parameter : currentProfile_        : Current profile
// -------------------------------------------------------------------------------------------------
void CScanningProfilesUser::ScanningProfilesSet( const TWStrings&    profileCmdArgNames_,
                                                 const TWStrings&    profileDisplayNames_,
                                                 const TWStrings&    profileAddtionalInfos_,
                                                 const std::wstring& currentProfile_ )
{
    m_profileCmdArgNames    = profileCmdArgNames_;
    m_profileDisplayNames   = profileDisplayNames_;
    m_profileAddtionalInfos = profileAddtionalInfos_;
    m_currentProfile        = currentProfile_;

    UpdateProfiles();
}

// =================================================================================================
//   Resets (clear) available scanning profiles.
// -------------------------------------------------------------------------------------------------
void CScanningProfilesUser::ScanningProfilesReset()
{
    ScanningProfilesSet( TWStrings(), TWStrings(), TWStrings(), L"" );
}

File: include/PolyWorksSDK/samples/plugins/common/CSharp/CSharpLineScanFramework/CScanningProfilesUser.cs
// =================================================================================================
//   CScanningProfilesUser class
//
// Copyright © InnovMetric Software Inc. 2018 All Rights Reserved
// -------------------------------------------------------------------------------------------------

using System.Collections.Generic;

namespace CompanyNameCsLineScanner
{

// =================================================================================================
//   Class used to manage scanning profiles.
// -------------------------------------------------------------------------------------------------
abstract class CScanningProfilesUser
{
    protected List< string > m_profileCmdArgNames;
    protected List< string > m_profileDisplayNames;
    protected List< string > m_profileAdditionalInfos;
    protected string         m_currentProfile;

    // =================================================================================================
    //   Sets available scanning profiles.
    //
    // Parameter : profileCmdArgNames_    : Profile names used for command arguments
    // Parameter : profileDisplayNames_   : Profile display names
    // Parameter : profileAddtionalInfos_ : Profile additional information
    // Parameter : currentProfile_        : Current profile
    // -------------------------------------------------------------------------------------------------
    public void ScanningProfilesSet( System.Array profileCmdArgNames,
                                     System.Array profileDisplayNames_,
                                     System.Array profileAdditionalInfos_,
                                     string       currentProfile_ )
    {
        m_profileCmdArgNames     = new List<string>();
        m_profileDisplayNames    = new List<string>();
        m_profileAdditionalInfos = new List<string>();
        m_currentProfile         = currentProfile_;

        foreach( string name in profileCmdArgNames )
        { 
            m_profileCmdArgNames.Add( name );
        }
        foreach( string name in profileDisplayNames_ )
        { 
            m_profileDisplayNames.Add( name );
        }
        foreach( string name in profileAdditionalInfos_ )
        { 
            m_profileAdditionalInfos.Add( name );
        }

        UpdateProfiles();
    }

    // =================================================================================================
    //   Gets available scanning profiles.
    //
    // Parameter : profileCmdArgNames_    : Profile names used for command arguments
    // Parameter : profileDisplayNames_   : Profile display names
    // Parameter : profileAddtionalInfos_ : Profile additional information
    // Parameter : currentProfile_        : Current profile
    // -------------------------------------------------------------------------------------------------
    public void ScanningProfilesGet( out List< string > profileCmdArgNames_,
                                     out List< string > profileDisplayNames_,
                                     out List< string > profileAdditionalInfos_,
                                     out string         currentProfiles_ )
    {
        profileCmdArgNames_     = m_profileCmdArgNames;
        profileDisplayNames_    = m_profileDisplayNames;
        profileAdditionalInfos_ = m_profileAdditionalInfos;
        currentProfiles_        = m_currentProfile;
    }

    public abstract void UpdateProfiles();
}

}